### ISSUE TITLE
[bitnami/redis] Redis slave auth is broken

### ISSUE THREAD

**Issue by adrafiq** [bitnami/charts#2510]
**bitnami/redis**:
This issue concerns the bitnami redis chart. Latest version => https://github.com/bitnami/charts/tree/master/bitnami/redis

**Describe the bug**
When you create a chart with master-slave replication on gke having password imported from existing secret. redis-slave cannot authenticate to master and logs show. Redis-master itself works and I am able to make connections with auth to it from all of telnet, redis-cli and node. Using the same password from secret.  
`unable to AUTH to MASTER: -ERR invalid password`

**To Reproduce**
1) Create a chart with following configuration. 
```
## Cluster settings
cluster:
  enabled: true
  slaveCount: 1

# Redis Master parameters
master:
  command: "/run.sh"
  configmap:
  ## Redis additional command line flags
  extraFlags: []
  ## Comma-separated list of Redis commands to disable
  disableCommands:
  - FLUSHDB
  - FLUSHALL

  ## Redis Master additional pod labels and annotations
  ## ref: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/
  podLabels: {}
  podAnnotations: {}

  ## Redis Master resource requests and limits
  ## ref: http://kubernetes.io/docs/user-guide/compute-resources/
  resources:
    requests:
      memory: 1Gi
      cpu: 256m
    limits: 
      memory: 2Gi
      cpu: 512m
  livenessProbe:
    enabled: true
    successThreshold: 1
    failureThreshold: 5
  readinessProbe:
    enabled: true
    successThreshold: 1
    failureThreshold: 5

  ## Redis Master Service properties
  service:
    ##  Redis Master Service type
    type: ClusterIP
    port: 6379
    annotations: {}
    labels: {}

  ## Persistent Volume
  persistence:
    enabled: true
    path: /data
    subPath: ""
    storageClass: "standard"
    accessModes:
    - ReadWriteOnce
    size: 10Gi
    ## Persistent Volume selectors
    ## https://kubernetes.io/docs/concepts/storage/persistent-volumes/#selector
    ## Currently handled by namespace
    # matchLabels: {}
    # matchExpressions: {}

## Redis Slave properties
slave:
  service:
    type: ClusterIP
    port: 6379
    annotations: {}
    labels: {}
  ## Persistent Volume
  persistence:
    enabled: false
    path: /data
    subPath: ""
    matchLabels: {
      release: redis-cache,
      role: master
    }
## Prometheus Exporter / Metrics
##
metrics:
  enabled: true
  image:
    registry: docker.io
    repository: bitnami/redis-exporter
    tag: 1.5.3-debian-10-r14
    pullPolicy: IfNotPresent
  podAnnotations:
    prometheus.io/scrape: "true"
    prometheus.io/port: "9121"
  service:
    type: ClusterIP

```
2) Check the logs of redis container on slave pod. 
`kubectl logs -f $SLAVE-POD -n $NAME_SPACE -c redis`
There will be logs similar to 
```
1:S 05 May 2020 05:08:46.690 * Connecting to MASTER redis-cache-master-0.redis-cache-headless.caching.svc.cluster.local:6379
1:S 05 May 2020 05:08:46.692 * MASTER <-> REPLICA sync started
1:S 05 May 2020 05:08:46.692 * Non blocking connect for SYNC fired the event.
1:S 05 May 2020 05:08:46.692 * Master replied to PING, replication can continue...
1:S 05 May 2020 05:08:46.692 # Unable to AUTH to MASTER: -ERR invalid password
```

**Expected behavior**
redis-slave should be authenticated to redis-master.

**Version of Helm and Kubernetes**:

`- Output of `helm version`:`

```
Client: &version.Version{SemVer:"v2.16.1", GitCommit:"bbdfe5e7803a12bbdf97e94cd847859890cf4050", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.16.1", GitCommit:"bbdfe5e7803a12bbdf97e94cd847859890cf4050", GitTreeState:"clean"}
```

- Output of `kubectl version`:

```
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.2", GitCommit:"52c56ce7a8272c798dbc29846288d7cd9fbae032", GitTreeState:"clean", BuildDate:"2020-04-21T01:25:41Z", GoVersion:"go1.13.10", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"15+", GitVersion:"v1.15.9-gke.24", GitCommit:"39e41a8d6b7221b901a95d3af358dea6994b4a40", GitTreeState:"clean", BuildDate:"2020-02-29T01:24:35Z", GoVersion:"go1.12.12b4", Compiler:"gc", Platform:"linux/amd64"}
```

**Additional context**
I suspect that for stateful spec generated for redis-slave could be problematic. 
```
- command:
    - /bin/bash
    - -c
    - |
      if [[ -n $REDIS_PASSWORD_FILE ]]; then
        password_aux=`cat ${REDIS_PASSWORD_FILE}`
        export REDIS_PASSWORD=$password_aux
      fi
      if [[ -n $REDIS_MASTER_PASSWORD_FILE ]]; then
        password_aux=`cat ${REDIS_MASTER_PASSWORD_FILE}`
        export REDIS_MASTER_PASSWORD=$password_aux
      fi
      if [[ ! -f /opt/bitnami/redis/etc/replica.conf ]];then
        cp /opt/bitnami/redis/mounted-etc/replica.conf /opt/bitnami/redis/etc/replica.conf
      fi
      if [[ ! -f /opt/bitnami/redis/etc/redis.conf ]];then
        cp /opt/bitnami/redis/mounted-etc/redis.conf /opt/bitnami/redis/etc/redis.conf
      fi
      ARGS=("--port" "${REDIS_PORT}")
      ARGS+=("--slaveof" "${REDIS_MASTER_HOST}" "${REDIS_MASTER_PORT_NUMBER}")
      ARGS+=("--protected-mode" "no")
      ARGS+=("--include" "/opt/bitnami/redis/etc/redis.conf")
      ARGS+=("--include" "/opt/bitnami/redis/etc/replica.conf")
      ARGS+=("--replicaof redis-cache-master-0.redis-cache-headless.caching.svc.cluster.local 6379")
      /run.sh "${ARGS[@]}"
```
Configuration flag `slaveof`  is depracated in favor of `replicaof` as of version 5.0 as described here. [SlaveOf Deprecation Disclaimer](https://redis.io/commands/slaveof) The chart generated must have `replicaof` flag. But not sure if this is causing the issue of auth failure. The current `replicaof` flag is added manually

**Comment by andresbono** [comment]
Thanks for opening this issue, @adrafiq. We will try to reproduce your issue and we will get back to you.

Did you identify this issue just with the latest version? Did you, by chance, try others?

**Comment by andresbono** [comment]
Hi @adrafiq, we have just released a new major version of Redis (`6.0`) can you give it a try and let us know if the error still persists?

**Comment by adrafiq** [comment]
> Thanks for opening this issue, @adrafiq. We will try to reproduce your issue and we will get back to you.
> 
> Did you identify this issue just with the latest version? Did you, by chance, try others?

I did reply every version below 5 and they seem to be working well on auth part. I tried several versions of 4.

**Comment by adrafiq** [comment]
> Hi @adrafiq, we have just released a new major version of Redis (`6.0`) can you give it a try and let us know if the error still persists?

Hey @andresbono .I can confirm the same issue on version 6.0.

**Comment by adrafiq** [comment]
> > Hi @adrafiq, we have just released a new major version of Redis (`6.0`) can you give it a try and let us know if the error still persists?
> 
> Hey @andresbono .I can confirm the same issue on version 6.0.
> Although the message has shifted
> from
> 
> ```
> Unable to AUTH to MASTER: -ERR invalid password
> ```
> 
> to
> 
> ```
> Unable to AUTH to MASTER: -WRONGPASS invalid username-password pair
> ```

**Comment by adrafiq** [comment]
I can report the statefulSpec generated still has `--slaveof` flag instead of `--replicaof`

**Comment by andresbono** [comment]
Hi @adrafiq, we couldn't reproduce your issue yet, it just works properly for us. You mention that you are using an existing secret to store the password:

> having password imported from existing secret.

However, I don't see that you are using the `existingSecret` parameters in the values that you are using. Is there something that I miss? Also, why did you close the issue?

---

> I can report the statefulSpec generated still has --slaveof flag instead of --replicaof
That's true, I can confirm it as well. Would you like to open a PR to solve the issue? Contributions are more than welcome!

**Comment by ukanga** [comment]
I did encounter the exact same error message, `Unable to AUTH to MASTER: -WRONGPASS invalid username-password pair`.  It turned out the issue was how I generated the password file used to create the secret. In my case, the file had a newline character. Hence, generating the file without a newline character did resolve the issue for me. 
It seems that Redis master pod does include the newline character in creating the credentials but that is stripped away by the Redis slave pods.
